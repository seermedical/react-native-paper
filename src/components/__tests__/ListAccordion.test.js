import * as React from 'react';
import { StyleSheet } from 'react-native';
import renderer from 'react-test-renderer';
import ListAccordion from '../List/ListAccordion.tsx';
import ListItem from '../List/ListItem.tsx';
import ListIcon from '../List/ListIcon.tsx';
import { red500 } from '../../styles/colors';

const styles = StyleSheet.create({
  coloring: {
    color: red500,
  },
});

it('renders list accordion with children', () => {
  const tree = renderer
    .create(
      <ListAccordion
        left={(props) => <ListIcon {...props} icon="folder" />}
        title="Expandable list item"
      >
        <ListItem title="First Item" />
      </ListAccordion>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders multiline list accordion', () => {
  const tree = renderer
    .create(
      <ListAccordion
        title="Expandable list item"
        description="Describes the expandable list item"
      >
        <ListItem title="List item 1" />
      </ListAccordion>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list accordion with left items', () => {
  const tree = renderer
    .create(
      <ListAccordion
        left={(props) => <ListIcon {...props} icon="star" />}
        title="Accordion item 1"
      >
        <ListItem
          left={(props) => <ListIcon {...props} icon="thumb-up" />}
          title="List item 1"
        />
      </ListAccordion>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders expanded accordion', () => {
  const tree = renderer
    .create(
      <ListAccordion title="Accordion item 1" expanded>
        <ListItem title="List item 1" />
      </ListAccordion>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list accordion with custom title and description styles', () => {
  const tree = renderer
    .create(
      <ListAccordion
        title="Accordion item 1"
        description="Describes the expandable list item"
        titleStyle={styles.coloring}
        descriptionStyle={styles.coloring}
      >
        <ListItem title="List item 1" />
      </ListAccordion>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
