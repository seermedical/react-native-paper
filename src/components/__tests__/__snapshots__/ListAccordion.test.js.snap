// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders expanded accordion 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#f6f6f6",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "expanded": true,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "padding": 8,
            },
            undefined,
          ],
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#6200ee",
                  },
                  undefined,
                ],
              ]
            }
          >
            Accordion item 1
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              undefined,
            ]
          }
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "marginVertical": 6,
              "paddingLeft": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
                undefined,
              ],
            ]
          }
        >
          List item 1
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders list accordion with children 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#f6f6f6",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "expanded": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "padding": 8,
            },
            undefined,
          ],
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "margin": 8,
                "width": 40,
              },
              undefined,
            ]
          }
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                  },
                  undefined,
                ],
              ]
            }
          >
            Expandable list item
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              undefined,
            ]
          }
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders list accordion with custom title and description styles 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#f6f6f6",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "expanded": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "padding": 8,
            },
            undefined,
          ],
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                  },
                  Object {
                    "color": "#f44336",
                  },
                ],
              ]
            }
          >
            Accordion item 1
          </Text>
          <Text
            numberOfLines={2}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 14,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                  },
                  Object {
                    "color": "#f44336",
                  },
                ],
              ]
            }
          >
            Describes the expandable list item
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders list accordion with left items 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#f6f6f6",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "expanded": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "padding": 8,
            },
            undefined,
          ],
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "margin": 8,
                "width": 40,
              },
              undefined,
            ]
          }
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                  },
                  undefined,
                ],
              ]
            }
          >
            Accordion item 1
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              undefined,
            ]
          }
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders multiline list accordion 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#f6f6f6",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "expanded": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "padding": 8,
            },
            undefined,
          ],
        ]
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              Object {
                "flex": 1,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 16,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                  },
                  undefined,
                ],
              ]
            }
          >
            Expandable list item
          </Text>
          <Text
            numberOfLines={2}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "fontSize": 14,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                  },
                  undefined,
                ],
              ]
            }
          >
            Describes the expandable list item
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "margin": 8,
              },
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontSize": 24,
                },
              ]
            }
          >
            □
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
